code/modules/organs/blood.dm

```diff
@@ -203,19 +203,30 @@
 		adjust_blood(amount, get_blood_data())
 	return amount
 
-/mob/living/carbon/proc/get_blood_data()
+/mob/living/proc/get_blood_data()
+
 	var/data = list()
-	data["donor"] = weakref(src)
-	data["blood_DNA"] = dna.unique_enzymes
-	data["blood_type"] = dna.b_type
-	data["species"] = species.name
-	data["has_oxy"] = species.blood_oxy
+	data["donor"]      = weakref(src)
+	data["blood_DNA"]  = get_unique_enzymes()
+	data["blood_type"] = get_blood_type()
+	data["species"]    = get_species_name()
+
 	var/list/temp_chem = list()
 	for(var/R in reagents.reagent_volumes)
 		temp_chem[R] = REAGENT_VOLUME(reagents, R)
-	data["trace_chem"] = temp_chem
-	data["dose_chem"] = chem_doses ? chem_doses.Copy() : list()
-	data["blood_color"] = species.get_blood_color(src)
+	data["trace_chem"]  = temp_chem
+	data["dose_chem"]   = chem_doses ? chem_doses.Copy() : list()
+
+	var/decl/species/my_species = get_species()
+	if(my_species)
+		data["has_oxy"]     = my_species.blood_oxy
+		data["blood_color"] = my_species.get_blood_color(src)
+	else if(isSynthetic())
```

Comment:
What's with this new use of `isSynthetic()`? Is `species.blood_oxy` not good enough? Note that there *are no* FBPs; blood color, oxygenation, etc. are already set on species like utility frames. This just makes it so you can't change the blood color of synthetic species at all.

EDIT: I just noticed this is only if species is false, which... is a bit odd, but I guess there's no problem with it. It won't clobber species data, at least.